### PR TITLE
update collector workflow to publish a prod layer

### PR DESCRIPTION
**Pull request by codeboten**
Also adding a fairly minimal release document.

### CODE REVIEW

File: .github/workflows/release-layer-collector.yml [hunk 1]
```diff
@@ -49,27 +49,27 @@ jobs:
...
+         - ap-southeast-2
+         - ca-central-1
+         - eu-central-1
+         - eu-north-1
+         - eu-west-1
+         - eu-west-2
+         - eu-west-3
+         - sa-east-1
+         - us-east-1
+         - us-east-2
```
Comment (tylerbenson) [inline .github/workflows/release-layer-collector.yml]: Indentation inconsistency with below.

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# OpenTelemetry Lambda Layer Release Procedure
+
+The release process is almost entirely managed by [GitHub actions](https://github.com/open-telemetry/opentelemetry-lambda/tree/main/.github/workflows). To publish a new layer:
+
+1. Create a new tag for the layer to publish. For example, to create a new collector layer, the following command is used:
+        `git tag layer-collector/0.0.8`
+2. Push the tag to [opentelemetry-lambda](https://github.com/open-telemetry/opentelemetry-lambda) repository to trigger the publish action.
```
Comment (tylerbenson) [inline RELEASE.md:7]: Should we encourage creating a github release with the proper ARN's for each region?  (We should probably automate this somehow.)

**Review by tylerbenson — APPROVED** [review]
Approved assuming indentation is fixed.

File: .github/workflows/release-layer-collector.yml (hunk 1 above)
Comment (codeboten) [reply .github/workflows/release-layer-collector.yml]: Fixed, thanks for catching!

File: RELEASE.md (hunk 2 above)
Comment (codeboten) [reply RELEASE.md:7]: Yeah i wasn't clear what the process should be. If we define it we can add it in the doc in a follow up

**Pull request merged**